src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -113,10 +112,10 @@ class CompilationState( statements : DeepDiveLog.Program, config : DeepDiveLog.C
         }
         if (decl.isQuery) variableTableNames += r
       }
-      case ExtractionRule(_,_) => ()
-      case InferenceRule(_,_,_,_) => ()
       case fdecl : FunctionDeclaration => function_schema += {fdecl.functionName -> fdecl}
       case f: FunctionCallRule => functionCallList += f
+      case f: InferenceRule => inferenceRules += f
```

Comment:
Would it be easy to get rid of these `ListBuffer`s using `collect { case f: Type => f }`?